File: 8_kyu/Sum_of_differences_in_array.lua
-- Calculates the range of the numbers in the array.
-- Returns distance between maximal and minimal value.
local function sum_of_differences(arr)
  if #arr == 0 then
    return 0
  end
  min = arr[1]
  max = arr[1]
  for _,v in pairs(arr) do
    if max < v then
      max = v
    elseif min > v then
      min = v
    end
  end
  return max - min
end

return sum_of_differences

File: 8_kyu/Sum_of_differences_in_array.rs
// Returns sum of all differences between ordered pairs of arr.
// = Returns width of the array
// = Returns max-min
// Returns None if array has less than 2 items.
fn sum_of_differences(arr: &[i8]) -> Option<i8> {
    if arr.len() > 1 {
        Some(arr.iter().max().unwrap() - arr.iter().min().unwrap())
    } else {
        None
    }
}
